Pin git dependencies (clack) — unpinned git deps cause build drift on main

## Context

While comparing \`main\` vs. the PR #349 branch I found that \`main\` currently **doesn't compile** on a fresh \`cargo check -p streamlib\`:

\`\`\`
error[E0308]: mismatched types
  --> libs/streamlib/src/core/clap/host.rs:408:33
   |
   |                           &mut display_buf,
   |                           ^^^^^^^^^^^^^^^^ expected \`&mut [u8]\`, found \`&mut Vec<MaybeUninit<u8>>\`
\`\`\`

The same file compiles cleanly on the #349 branch. Diffing Cargo.lock between the two:

\`\`\`
< source = \"git+https://github.com/prokopyl/clack#4541f03792635dfdc6be9441a01b786e9b4ccf18\"
> source = \"git+https://github.com/prokopyl/clack#f53f8fc132c001a332c0a7221267812dfb654e92\"
\`\`\`

The \`clack\` git dep is unpinned (tracks the branch HEAD, not a specific commit). When #349's work ran \`cargo fix --edition\` it fetched the newer commit, which introduces a \`value_to_text\` signature change that \`host.rs\` happens to match. Anyone doing a fresh clone of \`main\` today gets an older \`clack\` and a broken build.

## Root cause

\`libs/streamlib/Cargo.toml\` has:

\`\`\`toml
clack-host = { git = \"https://github.com/prokopyl/clack\" }
clack-plugin = { git = \"https://github.com/prokopyl/clack\" }
\`\`\`

No \`rev\` or \`tag\`. Cargo resolves this by fetching whatever's at HEAD of the default branch when the lockfile is regenerated.

## Task

1. Pin \`clack-host\` / \`clack-plugin\` to the specific commit that \`main\`'s code currently relies on (or upgrade to the commit that #349 pulled, since that's the one \`host.rs\` was written against):
   \`\`\`toml
   clack-host = { git = \"...\", rev = \"f53f8fc132c001a332c0a7221267812dfb654e92\" }
   clack-plugin = { git = \"...\", rev = \"f53f8fc132c001a332c0a7221267812dfb654e92\" }
   \`\`\`
2. Audit the rest of the workspace for any other unpinned git deps — \`grep -rn \"git =\" --include=Cargo.toml\`.
3. Document in CONTRIBUTING or CLAUDE.md: git deps must always be pinned (\`rev\` or \`tag\`).

## Priority

**High** — main is currently broken for fresh clones. Any new contributor hits this.

## Parent

None (infrastructure).

---

<!-- amos:ai-notes-begin -->
## AI Agent Notes

See the GitHub issue for full context.

## Priority

high

## Parent

#322 / #319 umbrella (for capability-split-related) or infrastructure.
<!-- amos:ai-notes-end -->

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -253,6 +253,9 @@ Run `cargo doc -p streamlib --no-deps` - fix any unresolved link warnings.
 - **macOS/iOS code**: `libs/streamlib/src/apple/`
 - **DO NOT** use `#[cfg]` inside platform-specific directories (already conditionally compiled)
 
+### Dependencies
+- **Git dependencies must be pinned** with `rev = "<commit sha>"` (or `tag = "..."`). Never use a bare `git = "..."` or `branch = "..."` — Cargo resolves those against the remote's current HEAD, so fresh clones drift out of sync with the lockfile and stop compiling. This applies to every `Cargo.toml` in the workspace, including `[patch.crates-io]` entries.
+
 ### Vulkan RHI Boundary — ABSOLUTE RULE
 
 **NOTHING outside the RHI (`vulkan/rhi/`) may touch Vulkan APIs directly.** No processor, utility, codec wrapper, or any other code may call `ash::Device`, `vkAllocateMemory`, `vkCreateImage`, or any Vulkan function without going through the RHI. This is non-negotiable.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,8 +84,8 @@ objc2-vision = "0.3.2"
 libc = "0.2.177"
 
 # Vulkan bindings (tatolab fork for VMA 3.3.0 patch)
-vulkanalia = { git = "https://github.com/tatolab/vulkanalia.git", branch = "tatolab/update-vma-3.3.0-patched", features = ["libloading", "provisional", "window"] }
-vulkanalia-vma = { git = "https://github.com/tatolab/vulkanalia.git", branch = "tatolab/update-vma-3.3.0-patched" }
+vulkanalia = { git = "https://github.com/tatolab/vulkanalia.git", rev = "982d32d293e5425753d595978776ee4fa4ded3a1", features = ["libloading", "provisional", "window"] }
+vulkanalia-vma = { git = "https://github.com/tatolab/vulkanalia.git", rev = "982d32d293e5425753d595978776ee4fa4ded3a1" }
 
 # Shader compilation
 # TODO: Add when implementing HLSL shader compilation:
@@ -114,5 +114,5 @@ streamlib-telemetry = { path = "libs/streamlib-telemetry" }
 # yuv = "0.8"
 
 [patch.crates-io]
-vulkanalia = { git = "https://github.com/tatolab/vulkanalia.git", branch = "tatolab/update-vma-3.3.0-patched" }
-vulkanalia-sys = { git = "https://github.com/tatolab/vulkanalia.git", branch = "tatolab/update-vma-3.3.0-patched" }
+vulkanalia = { git = "https://github.com/tatolab/vulkanalia.git", rev = "982d32d293e5425753d595978776ee4fa4ded3a1" }
+vulkanalia-sys = { git = "https://github.com/tatolab/vulkanalia.git", rev = "982d32d293e5425753d595978776ee4fa4ded3a1" }

--- a/libs/streamlib/Cargo.toml
+++ b/libs/streamlib/Cargo.toml
@@ -65,8 +65,8 @@ toml.workspace = true
 schemars = "0.8"  # JSON Schema generation from Rust types
 
 # CLAP audio plugin hosting
-clack-host = { git = "https://github.com/prokopyl/clack" }
-clack-extensions = { git = "https://github.com/prokopyl/clack", features = ["clack-host", "params"] }
+clack-host = { git = "https://github.com/prokopyl/clack", rev = "f53f8fc132c001a332c0a7221267812dfb654e92" }
+clack-extensions = { git = "https://github.com/prokopyl/clack", rev = "f53f8fc132c001a332c0a7221267812dfb654e92", features = ["clack-host", "params"] }
 
 # Audio resampling for real-time safe sample rate conversion
 rubato = "0.16"

--- a/plan/359-pin-clack-git-deps.md
+++ b/plan/359-pin-clack-git-deps.md
@@ -1,7 +1,7 @@
 ---
 whoami: amos
 name: "Pin git dependencies (clack) — main is currently broken for fresh clones"
-status: pending
+status: in_review
 description: '`clack-host` and `clack-plugin` are unpinned git deps (track branch HEAD, not commit). When the upstream moves, fresh clones of main get a different lockfile and stop compiling. Currently main fails to build `streamlib` on a fresh clone due to this drift. Pin to a specific rev + audit the workspace for other unpinned git deps + document in CLAUDE.md.'
 github_issue: 359
 adapters:

